File: android/src/main/java/com/lightstreamer/lightstreamer_flutter_client/MyClientListener.java
package com.lightstreamer.lightstreamer_flutter_client;

import com.lightstreamer.client.ClientListener;
import com.lightstreamer.client.LightstreamerClient;

import android.os.Handler;
import android.os.Looper;

import org.json.JSONException;
import org.json.JSONObject;

import io.flutter.plugin.common.BasicMessageChannel;

public class MyClientListener implements ClientListener {

    private BasicMessageChannel<String> _clientstatus_channel;
    final String clientId;

    public MyClientListener(BasicMessageChannel<String> clientstatus_channel, String clientId) {
        _clientstatus_channel = clientstatus_channel;
        this.clientId = clientId;
    }

    @Override
    public void onListenEnd() {
        // ...
    }

    @Override
    public void onListenStart() {
    }

    @Override
    public void onServerError(final int errorCode, final String errorMessage) {
        String json = toJson(new StringBuilder().append("ServerError:").append(errorCode).append(errorMessage).toString());
        try {
            new Handler(Looper.getMainLooper()).post(new Runnable() {
                @Override
                public void run() {
                    _clientstatus_channel.send(json);
                }
            });

        } catch (Exception e) {
            System.out.println("ERROR: " + e.getMessage());
        }
    }

    @Override
    public void onStatusChange(final String status) {
        String json = toJson(new StringBuilder().append("StatusChange:").append(status).toString());
        try {
            new Handler(Looper.getMainLooper()).post(new Runnable() {
                @Override
                public void run() {
                    _clientstatus_channel.send(json);
                }
            });

        } catch (Exception e) {
            System.out.println("ERROR: " + e.getMessage());
        }
    }

    @Override
    public void onPropertyChange(final String property) {
        String json = toJson(new StringBuilder().append("PropertyChange:").append(property).toString());

        try {
            new Handler(Looper.getMainLooper()).post(new Runnable() {
                @Override
                public void run() {
                    _clientstatus_channel.send(json);
                }
            });

        } catch (Exception e) {
            System.out.println("ERROR: " + e.getMessage());
        }
    }

    String toJson(String value) {
        try {
            JSONObject json = new JSONObject();
            json.put("id", clientId);
            json.put("value", value);
            return json.toString();
        } catch (JSONException e) {
            return null;
        }
    }
}

File: ios/Classes/MyClientListener.swift
import Foundation
import LightstreamerClient

class MyClientListener: ClientDelegate {
  let channel: FlutterBasicMessageChannel
  let clientId: String
  
  init(_ clientstatus_channel: FlutterBasicMessageChannel, _ clientId: String) {
    self.channel = clientstatus_channel
    self.clientId = clientId
  }
  
  func clientDidRemoveDelegate(_ client: LightstreamerClient) {}
  
  func clientDidAddDelegate(_ client: LightstreamerClient) {}
  
  func client(_ client: LightstreamerClient, didReceiveServerError errorCode: Int, withMessage errorMessage: String) {
    let json = toJson("ServerError:\(errorCode)\(errorMessage)")
    DispatchQueue.main.async {
      self.channel.sendMessage(json)
    }
  }
  
  func client(_ client: LightstreamerClient, didChangeStatus status: LightstreamerClient.Status) {
    let json = toJson("StatusChange:\(status)")
    DispatchQueue.main.async {
      self.channel.sendMessage(json)
    }
  }
  
  func client(_ client: LightstreamerClient, didChangeProperty property: String) {
    let json = toJson("PropertyChange:\(property)")
    DispatchQueue.main.async {
      self.channel.sendMessage(json)
    }
  }
  
  func toJson(_ value: String) -> String {
    let msg = ["id": clientId, "value": value]
    let encoder = JSONEncoder()
    let data = try! encoder.encode(msg)
    return String(data: data, encoding: .utf8)!
  }
}
